execution/executor/src/components/chunk_output.rs

```diff
@@ -89,13 +89,13 @@ impl ChunkOutput {
     }
 
     pub fn by_transaction_execution_sharded<V: VMExecutor>(
-        transactions: Vec<Transaction>,
+        block: Vec<SubBlocksForShard<Transaction>>,
         state_view: CachedStateView,
         maybe_block_gas_limit: Option<u64>,
     ) -> Result<Self> {
         let state_view_arc = Arc::new(state_view);
         let transaction_outputs = Self::execute_block_sharded::<V>(
-            transactions.clone(),
+            block.clone(),
```

Comment:
potential slow clone? I'm fine with it being here now.

Reply:
Hard to avoid the clone over here for now - this is similar in the unsharded path as well. This is something we should follow up to see if we can pass reference to the parallel execution instead.